File: SPbGY/4.java
package com.company;

import java.util.Scanner;

public class Main{

    public static void main(String[] args) {

        Scanner vvod = new Scanner(System.in);
        int a = vvod.nextInt();

        for (int g = 0; g < a; g ++){
            int mill = vvod.nextInt();
            String m = vvod.nextLine();
            String[] vars = m.split(" ", 3);
            String fir = vars[1];
            String sec = vars[2];

            int min_st = Math.min(fir.length(), sec.length());
            boolean f = false;
            for(int i = 0; i < min_st; i++){
                if (fir.charAt(i) != sec.charAt(i)){
                    f = true;
                    mill -= min_st - i;
                    mill -= sec.length() - i;
                    if ((mill >= 0)&&(mill % 2 == 0)){
                        System.out.println("Yes");
                    }
                    else{
                        System.out.println("No");
                    }
                    break;
                }
            }
            if(f == false){
                mill -= Math.max(fir.length(), sec.length()) - min_st;

                if ((mill >= 0)&&(mill % 2 == 0)){
                    System.out.println("Yes");
                }
                else{
                    System.out.println("No");
                }
            }
        }

    }

}

File: SPbGY/4.py
a = int(input())

def main_f():
  a, b, c = input().split(" ")
  a = int(a)

  if len(b) > len(c):
    max_st = b
    min_st = c
  else:
    max_st = c
    min_st = b
  
  if max_st.find(min_st) == 0:
    a -= len(max_st) - len(min_st)
    if a >= 0 and a % 2 == 0 :
      return "Yes"
    return "No"

  t = 0

  for i in range(len(min_st)):
    if min_st[i] != max_st[i]:
      t = i
      break
  
  a -= len(b) - t
  a -= len(c) - t

  if a >= 0 and (a % 2 == 0 or a == 0):
    return "Yes"
  return "No"
  

for i in range(a):
  print(main_f())
